kymatio/scattering3d/filter_bank.py

```diff
@@ -187,3 +188,219 @@ def solid_harmonic_3d(M, N, O, sigma, l, fourier=True):
     solid_harm *= norm_factor
 
     return solid_harm
+
+standard_orientations = {'cartesian' : [(1, 0, 0), (0, 1, 0), (0, 0, 1)]}
+
+def filter_bank(M, N, P, J, orientations='cartesian'):
+    if isinstance(orientations, str):
+        orientations = np.array(standard_orientations[orientations])
+    filters = {}
+    filters['psi'] = []
+
+    for j in range(J):
+        for orientation in orientations:
+            psi = {}
+            psi['j'] = j
+            psi['orientation'] = orientation
+
+            psi_signal = gabor_nd((M, N, P), orientation, j, sigma0=0.8,
+                    slant=3/len(orientations))
+                
+            psi_signal_fourier = np.fft.fftn(psi_signal)
+            # drop the imaginary part, it is zero anyway
+            psi_signal_fourier = np.real(psi_signal_fourier)
+            for res in range(min(j + 1, max(J - 1, 1))):
+                psi_signal_fourier_res = periodize_filter_fft(
+                    psi_signal_fourier, res)
+                psi[res] = psi_signal_fourier_res
+            filters['psi'].append(psi)
+
+    filters['phi'] = {}
+    phi_signal = gabor_nd((M, N, P), orientation, scale=J-1, xi0=0, sigma0=.8,
+            slant=1)
+    phi_signal_fourier = np.fft.fftn(phi_signal)
+    # drop the imaginary part, it is zero anyway
+    phi_signal_fourier = np.real(phi_signal_fourier)
+    filters['phi']['j'] = J
+    for res in range(J):
+        phi_signal_fourier_res = periodize_filter_fft(phi_signal_fourier, res)
+        filters['phi'][res] = phi_signal_fourier_res
+
+    return filters
+
+def periodize_filter_fft(x, res):
+    """
+        Parameters
+        ----------
+        x : numpy array
+            signal to periodize in Fourier
+        res :
+            resolution to which the signal is cropped.
+
+        Returns
+        -------
+        crop : numpy array
+            It returns a crop version of the filter, assuming that
+             the convolutions will be done via compactly supported signals.
+    """
+    M = x.shape[0]
+    N = x.shape[1]
+    P = x.shape[2]
+
+    crop = np.zeros((M // 2 ** res, N // 2 ** res, P // 2 ** res), x.dtype)
+
+    mask = np.ones(x.shape, np.float32)
+    len_x = int(M * (1 - 2 ** (-res)))
+    start_x = int(M * 2 ** (-res - 1))
+    len_y = int(N * (1 - 2 ** (-res)))
+    start_y = int(N * 2 ** (-res - 1))
+    len_z = int(P * (1 - 2 ** (-res)))
+    start_z = int(P * 2 ** (-res - 1))
+    mask[start_x:start_x + len_x,:, :] = 0
+    mask[:, start_y:start_y + len_y, :] = 0
+    mask[..., start_z:start_z + len_z] = 0
+
+    x = np.multiply(x,mask)
+
+    for k in range(int(M / 2 ** res)):
```

Comment:
OK, so this needs fixing. I believe that a simple

```python
def periodize(arr, res):
    ds = 2 ** res
    *b, d, h, w = arr.shape
    output = arr.reshape(b + [ds, d // ds, ds, h // ds, ds, w // ds]).sum(axis=(-6, -4, -2))
    return output
```

should give exactly the same output. (to be checked though)

Reply:
or is it doing something smart that I am missing?
I do recall that periodize_filter was doing something slightly different from periodize, in that it maybe disregarded higher frequencies. But actually I currently don't recall *why* one would want to do that

Reply:
Maybe @janden or @edouardoyallon knows or remembers something about this

Reply:
@edouardoyallon